spec/libs/twitter_processor_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'rails_helper'
+RSpec.describe TwitterProcessor do
+  let!(:company) { create(:company, twitter: 'https://twitter.com/netflix') }
+  let!(:processor) do
+    TwitterProcessor.new(company: company, number_of_tweets: 6)
+  end
+
+  describe 'screen name' do
+    context 'with twitter link valid' do
+      it 'screen name should be correct' do
+        expect(processor.screen_name).to eq('netflix')
+      end
+    end
+
+    context 'with twitter link empty' do
+      it 'screen name should be empty' do
+        company.update_columns(twitter: nil)
+        processor = TwitterProcessor.new(company: company, number_of_tweets: 6)
+
+        expect(processor.screen_name).to be(nil)
+      end
+    end
+  end
+
+  describe 'tweets' do
+    context 'working scrape' do
+      let!(:response_fixture) { Rails.root.join('spec/fixtures/tweets.json').to_s }
+      let!(:response_json) { Nokogiri::HTML(File.read(response_fixture)) }
```

Comment:
try to use `VCR`
al very least you'll be able to say that you worked with it :)

Reply:
mb `JSON.parse`?